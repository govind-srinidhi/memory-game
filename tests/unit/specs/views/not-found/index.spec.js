/**
 * @file It contains unit tests for not found.
 */
import { mount, createLocalVue } from "@vue/test-utils"
import Vuetify from "vuetify"
import NotFoundPage from "@/views/not-found"

let wrapper

const mountNotFoundPage = () => {
  const localVue = createLocalVue()

  localVue.use(Vuetify)

  localVue.prototype.$CONSTANTS = {
    LABELS: {
      NUMBER_OF_CARDS: "Number of cards",
      PLAY: "Play"
    },
    GAME_RESULTS: {
      CORRECT: "Correct",
      INCORRECT: "Incorrect"
    }
  }

  const vuetify = new Vuetify()

  wrapper = mount(NotFoundPage, {
    sync: false,
    localVue,
    vuetify
  })
}

const notFoundImage = () => wrapper.findComponent({ ref: "image_not_found" })
const notFoundHeader = () => wrapper.find(".not-found-header")

describe("home page", () => {
  beforeEach(() => {
    mountNotFoundPage()
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it("renders the not found page", () => {
    expect(wrapper).toBeTruthy()
  })

  it("renders elements correctly", async () => {
    expect(notFoundImage().exists()).toBeTruthy()
    expect(notFoundHeader().exists()).toBeTruthy()
  })

  it("matches the default snapshot", async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})