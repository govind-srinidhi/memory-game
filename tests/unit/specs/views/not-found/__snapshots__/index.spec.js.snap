// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page matches the default snapshot 1`] = `
<div class="container container--fluid">
  <div class="d-block text-center text-h4 not-found-header"></div>
  <div class="v-image v-responsive mt-12 theme--light" style="height: 650px;">
    <transition-stub name="fade-transition" mode="in-out">
      <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
    </transition-stub>
    <div class="v-responsive__content"></div>
  </div>
</div>
`;
